backend/adapter_processor/models.py

```diff
@@ -146,9 +149,11 @@ def get_context_window_size(self) -> int:
         adapter_metadata = self.get_adapter_meta_data()
         # Get the adapter_instance
         adapter_class = Adapterkit().get_adapter_class_by_adapter_id(self.adapter_id)
-        adapter_instance = adapter_class(adapter_metadata)
-        if isinstance(adapter_instance, LLMAdapter):
+        try:
+            adapter_instance = adapter_class(adapter_metadata)
             return adapter_instance.get_context_window_size()
+        except AdapterError as e:
+            logger.warning(f"Unable to retrieve context window size - {e}")
```

Comment:
@gaya3-zipstack is it okay to return 0 as the context window size in case of an error?

Reply:
Okay, I hope setting it to 0 doesn't cause any issues later. I don't have full context on where this value is used